django/http/multipartparser.py

```diff
@@ -166,15 +173,30 @@ def parse(self):
                 field_name = force_text(field_name, encoding, errors='replace')
 
                 if item_type == FIELD:
+                    # Avoid storing more than DATA_UPLOAD_MAX_NUMBER_FIELDS
+                    num_post_keys += 1
+                    if (settings.DATA_UPLOAD_MAX_NUMBER_FIELDS is not None and
+                            settings.DATA_UPLOAD_MAX_NUMBER_FIELDS < num_post_keys):
+                        raise SuspiciousOperation('Too many fields')
```

Comment:
Given that we never raise bare `SuspiciousOperation` anywhere else in Django's code, I wonder if we shouldn't also add subclasses to core/exceptions.py like we did with the rest. That said, even if we do not do that I'd write a bit more there to make sure that the developer immediately knows that this is about post data, maybe even mention `DATA_UPLOAD_MAX_NUMBER_FIELDS` in the message (do we have a policy or something for mentioning fixes in exceptions?)?